// @tamagui/text@1.144.4/dist/cjs/Emphasis.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: !0 });
}, __copyProps = (to, from, except, desc) => {
  if (from && typeof from == "object" || typeof from == "function")
    for (let key of __getOwnPropNames(from))
      !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: !0 }), mod);
var Emphasis_exports = {};
__export(Emphasis_exports, {
  Em: () => Em,
  Span: () => Span,
  Strong: () => Strong
});
module.exports = __toCommonJS(Emphasis_exports);
var import_web = require("@tamagui/web");
const Strong = (0, import_web.styled)(import_web.Text, {
  render: "strong",
  fontWeight: "bold"
}), Span = (0, import_web.styled)(import_web.Text, {
  render: "span"
}), Em = (0, import_web.styled)(import_web.Text, {
  render: "em",
  fontStyle: "italic"
});
