// @tamagui/web@1.144.4/dist/cjs/helpers/getTokenForKey.cjs
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
    for (var name in all) __defProp(target, name, {
      get: all[name],
      enumerable: !0
    });
  },
  __copyProps = (to, from, except, desc) => {
    if (from && typeof from == "object" || typeof from == "function") for (let key of __getOwnPropNames(from)) !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, {
      get: () => from[key],
      enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable
    });
    return to;
  };
var __toCommonJS = mod => __copyProps(__defProp({}, "__esModule", {
  value: !0
}), mod);
var getTokenForKey_exports = {};
__export(getTokenForKey_exports, {
  getLastFontFamilyToken: () => getLastFontFamilyToken,
  getTokenForKey: () => getTokenForKey,
  resolveVariableValue: () => resolveVariableValue,
  setLastFontFamilyToken: () => setLastFontFamilyToken
});
module.exports = __toCommonJS(getTokenForKey_exports);
var import_helpers = require("@tamagui/helpers"),
  import_config = require("../config.cjs"),
  import_createVariable = require("../createVariable.cjs"),
  import_getVariantExtras = require("./getVariantExtras.cjs");
const fontShorthand = {
  fontSize: "size",
  fontWeight: "weight"
};
let didLogMissingToken = !1,
  _lastFontFamilyToken = null;
function getLastFontFamilyToken() {
  return _lastFontFamilyToken;
}
function setLastFontFamilyToken(value) {
  _lastFontFamilyToken = value;
}
const getTokenForKey = (key, value, styleProps, styleState) => {
  let resolveAs = styleProps.resolveValues || "none";
  if (resolveAs === "none") return value;
  const {
      theme,
      conf = (0, import_config.getConfig)(),
      context,
      fontFamily,
      staticConfig
    } = styleState,
    themeValue = theme ? theme[value] || theme[value.slice(1)] : void 0,
    tokensParsed = conf.tokensParsed;
  let valOrVar,
    hasSet = !1;
  const customTokenAccept = staticConfig?.accept?.[key];
  if (customTokenAccept) {
    const val = themeValue ?? tokensParsed[customTokenAccept]?.[value];
    val != null && (resolveAs = "value", valOrVar = val, hasSet = !0);
  }
  if (themeValue) {
    if (resolveAs === "except-theme") return value;
    valOrVar = themeValue, process.env.NODE_ENV === "development" && styleState.debug === "verbose" && (globalThis.tamaguiAvoidTracking = !0, console.info(` - resolving ${key} to theme value ${value} resolveAs ${resolveAs}`, valOrVar), globalThis.tamaguiAvoidTracking = !1), hasSet = !0;
  } else {
    if (value in conf.specificTokens) hasSet = !0, valOrVar = conf.specificTokens[value];else {
      switch (key) {
        case "fontFamily":
          {
            valOrVar = (context?.language ? (0, import_getVariantExtras.getFontsForLanguage)(conf.fontsParsed, context.language) : conf.fontsParsed)[value]?.family || value, setLastFontFamilyToken(value), hasSet = !0;
            break;
          }
        case "fontSize":
        case "lineHeight":
        case "letterSpacing":
        case "fontWeight":
          {
            const fam = fontFamily || conf.defaultFontToken;
            if (fam) {
              const fontsParsed = context?.language ? (0, import_getVariantExtras.getFontsForLanguage)(conf.fontsParsed, context.language) : conf.fontsParsed;
              valOrVar = (fontsParsed[fam] || fontsParsed[conf.defaultFontToken])?.[fontShorthand[key] || key]?.[value] || value, hasSet = !0;
            }
            break;
          }
      }
      for (const cat in import_helpers.tokenCategories) if (key in import_helpers.tokenCategories[cat]) {
        const res = tokensParsed[cat]?.[value];
        res != null ? (valOrVar = res, hasSet = !0) : process.env.NODE_ENV === "development" && process.env.TAMAGUI_DISABLE_MISSING_TOKEN_LOG !== "1" && (didLogMissingToken || (didLogMissingToken = !0, console.groupCollapsed(`[tamagui] Warning: missing token ${key} in category ${cat} - ${value} (open for details)`), console.info(`Note: this could just be due to you not setting all the theme tokens Tamagui expects, which is harmless, but
                    it also often can be because you have a duplicated Tamagui in your bundle, which can cause tricky bugs.`), console.info(`To see if you have duplicated dependencies, in Chrome DevTools hit CMD+P and type TamaguiProvider.
                    If you see both a .cjs and a .mjs entry, it's duplicated.`), console.info("You can debug that issue by opening the .mjs and .cjs files and setting a breakpoint at the top of each."), console.info("We only log this warning one time as it's sometimes harmless, to disable this log entirely set process.env.TAMAGUI_DISABLE_MISSING_TOKEN_LOG=1."), console.groupEnd()));
      }
    }
    if (!hasSet) {
      const spaceVar = tokensParsed.space[value];
      spaceVar != null && (valOrVar = spaceVar, hasSet = !0);
    }
  }
  if (hasSet) {
    const out = resolveVariableValue(key, valOrVar, resolveAs);
    return process.env.NODE_ENV === "development" && styleState.debug === "verbose" && (globalThis.tamaguiAvoidTracking = !0, console.info("resolved", resolveAs, valOrVar, out), globalThis.tamaguiAvoidTracking = !1), out;
  }
  process.env.NODE_ENV === "development" && styleState.debug === "verbose" && console.warn(`Warning: no token found for ${key}, omitting`);
};
function resolveVariableValue(key, valOrVar, resolveValues) {
  if (resolveValues === "none") return valOrVar;
  if ((0, import_createVariable.isVariable)(valOrVar)) {
    if (resolveValues === "value") return valOrVar.val;
    const get = valOrVar?.get;
    return typeof get == "function" ? get(resolveValues === "web" ? "web" : void 0) : valOrVar.variable;
  }
  return valOrVar;
}
