// @tamagui/web@1.144.4/dist/cjs/helpers/getTokenForKey.native.js
"use strict";

var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
    for (var name in all) __defProp(target, name, {
      get: all[name],
      enumerable: !0
    });
  },
  __copyProps = (to, from, except, desc) => {
    if (from && typeof from == "object" || typeof from == "function") for (let key of __getOwnPropNames(from)) !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, {
      get: () => from[key],
      enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable
    });
    return to;
  };
var __toCommonJS = mod => __copyProps(__defProp({}, "__esModule", {
  value: !0
}), mod);
var getTokenForKey_exports = {};
__export(getTokenForKey_exports, {
  getLastFontFamilyToken: () => getLastFontFamilyToken,
  getTokenForKey: () => getTokenForKey,
  resolveVariableValue: () => resolveVariableValue,
  setLastFontFamilyToken: () => setLastFontFamilyToken
});
module.exports = __toCommonJS(getTokenForKey_exports);
var import_helpers = require("@tamagui/helpers"),
  import_config = require("../config.native.js"),
  import_createVariable = require("../createVariable.native.js"),
  import_getVariantExtras = require("./getVariantExtras.native.js"),
  fontShorthand = {
    fontSize: "size",
    fontWeight: "weight"
  },
  didLogMissingToken = !1,
  _lastFontFamilyToken = null;
function getLastFontFamilyToken() {
  return _lastFontFamilyToken;
}
function setLastFontFamilyToken(value) {
  _lastFontFamilyToken = value;
}
var getTokenForKey = function (key, value, styleProps, styleState) {
  var _staticConfig_accept,
    resolveAs = styleProps.resolveValues || "none";
  if (resolveAs === "none") return value;
  var {
      theme,
      conf = (0, import_config.getConfig)(),
      context,
      fontFamily,
      staticConfig
    } = styleState,
    themeValue = theme ? theme[value] || theme[value.slice(1)] : void 0,
    tokensParsed = conf.tokensParsed,
    valOrVar,
    hasSet = !1,
    customTokenAccept = staticConfig == null || (_staticConfig_accept = staticConfig.accept) === null || _staticConfig_accept === void 0 ? void 0 : _staticConfig_accept[key];
  if (customTokenAccept) {
    var _tokensParsed_customTokenAccept,
      val = themeValue ?? ((_tokensParsed_customTokenAccept = tokensParsed[customTokenAccept]) === null || _tokensParsed_customTokenAccept === void 0 ? void 0 : _tokensParsed_customTokenAccept[value]);
    val != null && (resolveAs = "value", valOrVar = val, hasSet = !0);
  }
  if (themeValue) {
    if (resolveAs === "except-theme") return value;
    valOrVar = themeValue, process.env.NODE_ENV === "development" && styleState.debug === "verbose" && (globalThis.tamaguiAvoidTracking = !0, console.info(` - resolving ${key} to theme value ${value} resolveAs ${resolveAs}`, valOrVar), globalThis.tamaguiAvoidTracking = !1), hasSet = !0;
  } else {
    if (value in conf.specificTokens) hasSet = !0, valOrVar = conf.specificTokens[value];else {
      switch (key) {
        case "fontFamily":
          {
            var _fontsParsed_value,
              fontsParsed = context?.language ? (0, import_getVariantExtras.getFontsForLanguage)(conf.fontsParsed, context.language) : conf.fontsParsed;
            valOrVar = ((_fontsParsed_value = fontsParsed[value]) === null || _fontsParsed_value === void 0 ? void 0 : _fontsParsed_value.family) || value, setLastFontFamilyToken(value), hasSet = !0;
            break;
          }
        case "fontSize":
        case "lineHeight":
        case "letterSpacing":
        case "fontWeight":
          {
            var fam = fontFamily || conf.defaultFontToken;
            if (fam) {
              var _font_,
                fontsParsed1 = context?.language ? (0, import_getVariantExtras.getFontsForLanguage)(conf.fontsParsed, context.language) : conf.fontsParsed,
                font = fontsParsed1[fam] || fontsParsed1[conf.defaultFontToken];
              valOrVar = (font == null || (_font_ = font[fontShorthand[key] || key]) === null || _font_ === void 0 ? void 0 : _font_[value]) || value, hasSet = !0;
            }
            break;
          }
      }
      for (var cat in import_helpers.tokenCategories) if (key in import_helpers.tokenCategories[cat]) {
        var _tokensParsed_cat,
          res = (_tokensParsed_cat = tokensParsed[cat]) === null || _tokensParsed_cat === void 0 ? void 0 : _tokensParsed_cat[value];
        res != null ? (valOrVar = res, hasSet = !0) : process.env.NODE_ENV === "development" && process.env.TAMAGUI_DISABLE_MISSING_TOKEN_LOG !== "1" && (didLogMissingToken || (didLogMissingToken = !0, console.groupCollapsed(`[tamagui] Warning: missing token ${key} in category ${cat} - ${value} (open for details)`), console.info(`Note: this could just be due to you not setting all the theme tokens Tamagui expects, which is harmless, but
                    it also often can be because you have a duplicated Tamagui in your bundle, which can cause tricky bugs.`), console.info(`To see if you have duplicated dependencies, in Chrome DevTools hit CMD+P and type TamaguiProvider.
                    If you see both a .cjs and a .mjs entry, it's duplicated.`), console.info("You can debug that issue by opening the .mjs and .cjs files and setting a breakpoint at the top of each."), console.info("We only log this warning one time as it's sometimes harmless, to disable this log entirely set process.env.TAMAGUI_DISABLE_MISSING_TOKEN_LOG=1."), console.groupEnd()));
      }
    }
    if (!hasSet) {
      var spaceVar = tokensParsed.space[value];
      spaceVar != null && (valOrVar = spaceVar, hasSet = !0);
    }
  }
  if (hasSet) {
    var out = resolveVariableValue(key, valOrVar, resolveAs);
    return process.env.NODE_ENV === "development" && styleState.debug === "verbose" && (globalThis.tamaguiAvoidTracking = !0, console.info("resolved", resolveAs, valOrVar, out), globalThis.tamaguiAvoidTracking = !1), out;
  }
  process.env.NODE_ENV === "development" && styleState.debug === "verbose" && console.warn(`Warning: no token found for ${key}, omitting`);
};
function resolveVariableValue(key, valOrVar, resolveValues) {
  if (resolveValues === "none") return valOrVar;
  if ((0, import_createVariable.isVariable)(valOrVar)) {
    if (resolveValues === "value") return valOrVar.val;
    var get = valOrVar?.get;
    if (key !== "shadowColor" && typeof get == "function") {
      var resolveDynamicFor = resolveValues === "web" ? "web" : void 0;
      return get(resolveDynamicFor);
    }
    return valOrVar.val;
  }
  return valOrVar;
}
