// @tamagui/image@1.144.4/dist/jsx/createImage.js
import { getTokenValue, styled } from "@tamagui/web";
import { jsx } from "react/jsx-runtime";
const defaultObjectFitMap = (objectFit) => {
  switch (objectFit) {
    case "fill":
      return "stretch";
    case "none":
      return "center";
    case "scale-down":
      return "contain";
    case "contain":
      return "contain";
    default:
      return "cover";
  }
}, defaultTransformSource = (props) => {
  const { src, source, width, height } = props;
  return source || (src && typeof src != "string" ? src : {
    uri: src,
    width,
    height
  });
};
function createImage(options) {
  const {
    Component,
    mapObjectFitToResizeMode = defaultObjectFitMap,
    resizeModePropName = "resizeMode",
    objectPositionPropName,
    transformSource = defaultTransformSource
  } = options, ImageComponent = styled(
    Component,
    {
      name: "Image"
    },
    {
      inlineProps: /* @__PURE__ */ new Set([
        "source",
        "placeholder",
        "transition",
        "contentFit",
        "contentPosition",
        "cachePolicy",
        "recyclingKey",
        "allowDownscaling",
        "autoplay",
        "blurRadius",
        "priority",
        "placeholderContentFit",
        "responsivePolicy",
        "onLoadStart",
        "onProgress",
        "onLoadEnd"
      ])
    }
  ).styleable((incomingProps, ref) => {
    const props = incomingProps, {
      src,
      width,
      height,
      objectFit,
      objectPosition,
      // web only props - filter out on native
      decoding,
      elementTiming,
      fetchpriority,
      isMap,
      loading,
      sizes,
      useMap,
      onLoad,
      onError,
      ...rest
    } = props, resolvedWidth = typeof width == "string" && width[0] === "$" ? getTokenValue(width) : width, resolvedHeight = typeof height == "string" && height[0] === "$" ? getTokenValue(height) : height, finalSource = transformSource({
      src,
      width: resolvedWidth,
      height: resolvedHeight
    }), finalProps = {
      ...rest,
      source: finalSource,
      style: {
        width: resolvedWidth,
        height: resolvedHeight
      }
    };
    return objectFit && (finalProps[resizeModePropName] = mapObjectFitToResizeMode(objectFit)), objectPositionPropName && objectPosition && (finalProps[objectPositionPropName] = objectPosition), onLoad && (finalProps.onLoad = (e) => {
      const source = e?.nativeEvent?.source || e?.source || {};
      onLoad({
        target: {
          naturalHeight: source?.height,
          naturalWidth: source?.width
        },
        type: "load"
      });
    }), onError && (finalProps.onError = () => {
      onError({
        type: "error"
      });
    }), /* @__PURE__ */ jsx(Component, { ref, ...finalProps });
  }), comp = Component;
  return ImageComponent.getSize = comp.getSize || (() => {
  }), ImageComponent.getSizeWithHeaders = comp.getSizeWithHeaders || (() => {
  }), ImageComponent.prefetch = comp.prefetch || (() => {
  }), ImageComponent.prefetchWithMetadata = comp.prefetchWithMetadata || (() => {
  }), ImageComponent.abortPrefetch = comp.abortPrefetch || (() => {
  }), ImageComponent.queryCache = comp.queryCache || (() => {
  }), ImageComponent;
}
export {
  createImage
};
