// @tamagui/image@1.144.4/dist/jsx/createImage.native.js
"use strict";

var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
    for (var name in all) __defProp(target, name, {
      get: all[name],
      enumerable: !0
    });
  },
  __copyProps = (to, from, except, desc) => {
    if (from && typeof from == "object" || typeof from == "function") for (let key of __getOwnPropNames(from)) !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, {
      get: () => from[key],
      enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable
    });
    return to;
  };
var __toCommonJS = mod => __copyProps(__defProp({}, "__esModule", {
  value: !0
}), mod);
var createImage_exports = {};
__export(createImage_exports, {
  createImage: () => createImage
});
module.exports = __toCommonJS(createImage_exports);
var import_jsx_runtime = require("react/jsx-runtime"),
  import_web = require("@tamagui/web"),
  defaultObjectFitMap = function (objectFit) {
    switch (objectFit) {
      case "fill":
        return "stretch";
      case "none":
        return "center";
      case "scale-down":
        return "contain";
      case "contain":
        return "contain";
      default:
        return "cover";
    }
  },
  defaultTransformSource = function (props) {
    var {
      src,
      source,
      width,
      height
    } = props;
    return source || (src && typeof src != "string" ? src : {
      uri: src,
      width,
      height
    });
  };
function createImage(options) {
  var {
      Component,
      mapObjectFitToResizeMode = defaultObjectFitMap,
      resizeModePropName = "resizeMode",
      objectPositionPropName,
      transformSource = defaultTransformSource
    } = options,
    inlinePropsSet = /* @__PURE__ */new Set(["source", "placeholder", "transition", "contentFit", "contentPosition", "cachePolicy", "recyclingKey", "allowDownscaling", "autoplay", "blurRadius", "priority", "placeholderContentFit", "responsivePolicy", "onLoadStart", "onProgress", "onLoadEnd"]),
    StyledImage = (0, import_web.styled)(Component, {
      name: "Image"
    }, {
      inlineProps: inlinePropsSet
    }),
    ImageComponent = StyledImage.styleable(function (incomingProps, ref) {
      var props = incomingProps,
        {
          src,
          width,
          height,
          objectFit,
          objectPosition,
          // web only props - filter out on native
          decoding,
          elementTiming,
          fetchpriority,
          isMap,
          loading,
          sizes,
          useMap,
          onLoad,
          onError,
          ...rest
        } = props,
        resolvedWidth = typeof width == "string" && width[0] === "$" ? (0, import_web.getTokenValue)(width) : width,
        resolvedHeight = typeof height == "string" && height[0] === "$" ? (0, import_web.getTokenValue)(height) : height,
        finalSource = transformSource({
          src,
          width: resolvedWidth,
          height: resolvedHeight
        }),
        finalProps = {
          ...rest,
          source: finalSource,
          style: {
            width: resolvedWidth,
            height: resolvedHeight
          }
        };
      return objectFit && (finalProps[resizeModePropName] = mapObjectFitToResizeMode(objectFit)), objectPositionPropName && objectPosition && (finalProps[objectPositionPropName] = objectPosition), onLoad && (finalProps.onLoad = function (e) {
        var _e_nativeEvent,
          source = (e == null || (_e_nativeEvent = e.nativeEvent) === null || _e_nativeEvent === void 0 ? void 0 : _e_nativeEvent.source) || e?.source || {};
        onLoad({
          target: {
            naturalHeight: source?.height,
            naturalWidth: source?.width
          },
          type: "load"
        });
      }), onError && (finalProps.onError = function () {
        onError({
          type: "error"
        });
      }), /* @__PURE__ */(0, import_jsx_runtime.jsx)(Component, {
        ref,
        ...finalProps
      });
    }),
    comp = Component;
  return ImageComponent.getSize = comp.getSize || function () {}, ImageComponent.getSizeWithHeaders = comp.getSizeWithHeaders || function () {}, ImageComponent.prefetch = comp.prefetch || function () {}, ImageComponent.prefetchWithMetadata = comp.prefetchWithMetadata || function () {}, ImageComponent.abortPrefetch = comp.abortPrefetch || function () {}, ImageComponent.queryCache = comp.queryCache || function () {}, ImageComponent;
}
