// @tamagui/themes@1.144.4/dist/cjs/defaultGrandChildrenThemes.native.js
"use strict";

var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
    for (var name in all) __defProp(target, name, {
      get: all[name],
      enumerable: !0
    });
  },
  __copyProps = (to, from, except, desc) => {
    if (from && typeof from == "object" || typeof from == "function") for (let key of __getOwnPropNames(from)) !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, {
      get: () => from[key],
      enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable
    });
    return to;
  };
var __toCommonJS = mod => __copyProps(__defProp({}, "__esModule", {
  value: !0
}), mod);
var defaultGrandChildrenThemes_exports = {};
__export(defaultGrandChildrenThemes_exports, {
  defaultGrandChildrenThemes: () => defaultGrandChildrenThemes
});
module.exports = __toCommonJS(defaultGrandChildrenThemes_exports);
var defaultGrandChildrenThemes = {
  accent: {
    template: "inverse"
  },
  alt1: {
    template: "alt1"
  },
  alt2: {
    template: "alt2"
  },
  surface1: {
    template: "surface1"
  },
  surface2: {
    template: "surface2"
  },
  surface3: {
    template: "surface3"
  }
};
