// @tamagui/themes@1.144.4/dist/cjs/defaultColorThemes.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: !0 });
}, __copyProps = (to, from, except, desc) => {
  if (from && typeof from == "object" || typeof from == "function")
    for (let key of __getOwnPropNames(from))
      !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: !0 }), mod);
var defaultColorThemes_exports = {};
__export(defaultColorThemes_exports, {
  colorThemesToExtra: () => colorThemesToExtra,
  defaultColorThemes: () => defaultColorThemes
});
module.exports = __toCommonJS(defaultColorThemes_exports);
var import_colors = require("@tamagui/colors"), import_v5_palettes = require("./v5-palettes");
function paletteToNamedColors(name, palette) {
  return {
    [`${name}1`]: palette[0],
    [`${name}2`]: palette[1],
    [`${name}3`]: palette[2],
    [`${name}4`]: palette[3],
    [`${name}5`]: palette[4],
    [`${name}6`]: palette[5],
    [`${name}7`]: palette[6],
    [`${name}8`]: palette[7],
    [`${name}9`]: palette[8],
    [`${name}10`]: palette[9],
    [`${name}11`]: palette[10],
    [`${name}12`]: palette[11]
  };
}
const defaultColorThemes = {
  black: {
    palette: {
      dark: Object.values(import_v5_palettes.darkPalette),
      light: Object.values(import_v5_palettes.darkPalette)
    }
  },
  white: {
    palette: {
      dark: Object.values(import_v5_palettes.lightPalette),
      light: Object.values(import_v5_palettes.lightPalette)
    }
  },
  gray: {
    palette: {
      dark: Object.values(import_colors.grayDark),
      light: Object.values(import_colors.gray)
    }
  },
  blue: {
    palette: {
      dark: Object.values(import_colors.blueDark),
      light: Object.values(import_colors.blue)
    }
  },
  red: {
    palette: {
      dark: Object.values(import_colors.redDark),
      light: Object.values(import_colors.red)
    }
  },
  yellow: {
    palette: {
      dark: Object.values(import_colors.yellowDark),
      light: Object.values(import_colors.yellow)
    }
  },
  green: {
    palette: {
      dark: Object.values(import_colors.greenDark),
      light: Object.values(import_colors.green)
    }
  },
  neutral: {
    palette: {
      dark: import_v5_palettes.neutralPalette,
      light: import_v5_palettes.neutralPalette
    }
  }
};
function colorThemesToExtra(colorThemes, scheme) {
  const extra = {};
  for (const [name, theme] of Object.entries(colorThemes)) {
    const palette = scheme === "light" ? theme.palette.light : theme.palette.dark, namedColors = paletteToNamedColors(name, palette);
    Object.assign(extra, namedColors);
  }
  return extra;
}
