// @tamagui/themes@1.144.4/dist/cjs/v5-palettes.native.js
"use strict";

var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
    for (var name in all) __defProp(target, name, {
      get: all[name],
      enumerable: !0
    });
  },
  __copyProps = (to, from, except, desc) => {
    if (from && typeof from == "object" || typeof from == "function") for (let key of __getOwnPropNames(from)) !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, {
      get: () => from[key],
      enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable
    });
    return to;
  };
var __toCommonJS = mod => __copyProps(__defProp({}, "__esModule", {
  value: !0
}), mod);
var v5_palettes_exports = {};
__export(v5_palettes_exports, {
  blackColors: () => blackColors,
  darkPalette: () => darkPalette,
  darkShadows: () => darkShadows,
  lightPalette: () => lightPalette,
  lightShadows: () => lightShadows,
  neutral: () => neutral,
  neutralPalette: () => neutralPalette,
  palettes: () => palettes,
  whiteBlack: () => whiteBlack,
  whiteColors: () => whiteColors
});
module.exports = __toCommonJS(v5_palettes_exports);
var import_colors = require("@tamagui/colors"),
  darkPalette = ["#050505", "#151515", "#191919", "#232323", "#282828", "#323232", "#424242", "#494949", "#545454", "#626262", "#a5a5a5", "#fff"],
  lightPalette = ["#fff", "#f8f8f8", "hsl(0, 0%, 93%)", "hsl(0, 0%, 87%)", "hsl(0, 0%, 80%)", "hsl(0, 0%, 74%)", "hsl(0, 0%, 68%)", "hsl(0, 0%, 60%)", "hsl(0, 0%, 48%)", "hsl(0, 0%, 38%)", "hsl(0, 0%, 20%)", "hsl(0, 0%, 2%)"],
  neutralPalette = ["hsl(0, 0%, 68%)", "hsl(0, 0%, 65%)", "hsl(0, 0%, 62%)", "hsl(0, 0%, 59%)", "hsl(0, 0%, 56%)", "hsl(0, 0%, 53%)", "hsl(0, 0%, 50%)", "hsl(0, 0%, 47%)", "hsl(0, 0%, 44%)", "hsl(0, 0%, 41%)", "hsl(0, 0%, 38%)", "hsl(0, 0%, 32%)"],
  whiteBlack = {
    white: "rgba(255,255,255,1)",
    white0: "rgba(255,255,255,0)",
    white02: "rgba(255,255,255,0.2)",
    white04: "rgba(255,255,255,0.4)",
    white06: "rgba(255,255,255,0.6)",
    white08: "rgba(255,255,255,0.8)",
    black: "rgba(0,0,0,1)",
    black0: "rgba(0,0,0,0)",
    black02: "rgba(0,0,0,0.2)",
    black04: "rgba(0,0,0,0.4)",
    black06: "rgba(0,0,0,0.6)",
    black08: "rgba(0,0,0,0.8)"
  },
  darkShadows = {
    shadow1: "rgba(0,0,0,0.1)",
    shadow2: "rgba(0,0,0,0.18)",
    shadow3: "rgba(0,0,0,0.25)",
    shadow4: "rgba(0,0,0,0.4)",
    shadow5: "rgba(0,0,0,0.55)",
    shadow6: "rgba(0,0,0,0.66)"
  },
  lightShadows = {
    shadow1: "rgba(0,0,0,0.025)",
    shadow2: "rgba(0,0,0,0.04)",
    shadow3: "rgba(0,0,0,0.06)",
    shadow4: "rgba(0,0,0,0.095)",
    shadow5: "rgba(0,0,0,0.195)",
    shadow6: "rgba(0,0,0,0.3)"
  },
  blackColors = {
    black1: darkPalette[0],
    black2: darkPalette[1],
    black3: darkPalette[2],
    black4: darkPalette[3],
    black5: darkPalette[4],
    black6: darkPalette[5],
    black7: darkPalette[6],
    black8: darkPalette[7],
    black9: darkPalette[8],
    black10: darkPalette[9],
    black11: darkPalette[10],
    black12: darkPalette[11]
  },
  whiteColors = {
    white1: lightPalette[0],
    white2: lightPalette[1],
    white3: lightPalette[2],
    white4: lightPalette[3],
    white5: lightPalette[4],
    white6: lightPalette[5],
    white7: lightPalette[6],
    white8: lightPalette[7],
    white9: lightPalette[8],
    white10: lightPalette[9],
    white11: lightPalette[10],
    white12: lightPalette[11]
  },
  neutral = {
    neutral1: neutralPalette[0],
    neutral2: neutralPalette[1],
    neutral3: neutralPalette[2],
    neutral4: neutralPalette[3],
    neutral5: neutralPalette[4],
    neutral6: neutralPalette[5],
    neutral7: neutralPalette[6],
    neutral8: neutralPalette[7],
    neutral9: neutralPalette[8],
    neutral10: neutralPalette[9],
    neutral11: neutralPalette[10],
    neutral12: neutralPalette[11]
  },
  palettes = {
    dark: darkPalette,
    light: lightPalette,
    dark_blue: Object.values(import_colors.blueDark),
    light_blue: Object.values(import_colors.blue),
    dark_gray: Object.values(import_colors.grayDark),
    light_gray: Object.values(import_colors.gray),
    dark_green: Object.values(import_colors.greenDark),
    light_green: Object.values(import_colors.green),
    dark_red: Object.values(import_colors.redDark),
    light_red: Object.values(import_colors.red),
    dark_yellow: Object.values(import_colors.yellowDark),
    light_yellow: Object.values(import_colors.yellow)
  };
