// @tamagui/config@1.144.4/dist/esm/v5-base.mjs
import { shorthands } from "@tamagui/shorthands/v5";
import { themes, tokens } from "@tamagui/themes/v5";
import { fonts } from "./v5-fonts.mjs";
import { media, mediaQueryDefaultActive } from "./v5-media.mjs";
import { shorthands as shorthands2 } from "@tamagui/shorthands/v5";
import { createThemes } from "@tamagui/theme-builder";
import { themes as themes2, tokens as tokens2, createV5Theme } from "@tamagui/themes/v5";
import { createSystemFont, fonts as fonts2 } from "./v5-fonts.mjs";
import { breakpoints, media as media2, mediaQueryDefaultActive as mediaQueryDefaultActive2 } from "./v5-media.mjs";
const selectionStyles = theme => theme.color5 ? {
    backgroundColor: theme.color5,
    color: theme.color11
  } : null,
  settings = {
    mediaQueryDefaultActive,
    defaultFont: "body",
    fastSchemeChange: !0,
    shouldAddPrefersColorThemes: !0,
    allowedStyleValues: "somewhat-strict-web",
    addThemeClassName: "html",
    onlyAllowShorthands: !0,
    styleCompat: "react-native"
  },
  defaultConfigBase = {
    media,
    shorthands,
    themes,
    tokens,
    fonts,
    selectionStyles,
    settings
  };
export { breakpoints, createSystemFont, createThemes, createV5Theme, defaultConfigBase, fonts2 as fonts, media2 as media, mediaQueryDefaultActive2 as mediaQueryDefaultActive, selectionStyles, settings, shorthands2 as shorthands, themes2 as themes, tokens2 as tokens };
