// @tamagui/config@1.144.4/dist/esm/v5.native.js
import { shorthands } from "@tamagui/shorthands/v5";
import { tokens, defaultThemes } from "@tamagui/themes/v5";
import { animations } from "./v5-animations.native.js";
import { fonts } from "./v5-fonts.native.js";
import { media, mediaQueryDefaultActive } from "./v5-media.native.js";
import { shorthands as shorthands2 } from "@tamagui/shorthands/v5";
import { createThemes } from "@tamagui/theme-builder";
import { tokens as tokens2, defaultThemes as defaultThemes2 } from "@tamagui/themes/v5";
import { animations as animations2 } from "./v5-animations.native.js";
import { createSystemFont, fonts as fonts2 } from "./v5-fonts.native.js";
import { breakpoints, media as media2, mediaQueryDefaultActive as mediaQueryDefaultActive2 } from "./v5-media.native.js";
var selectionStyles = function (theme) {
    return theme.color5 ? {
      backgroundColor: theme.color5,
      color: theme.color11
    } : null;
  },
  settings = {
    mediaQueryDefaultActive,
    defaultFont: "body",
    fastSchemeChange: !0,
    shouldAddPrefersColorThemes: !0,
    allowedStyleValues: "somewhat-strict-web",
    themeClassNameOnRoot: !0,
    onlyAllowShorthands: !0,
    // allow two inverses (tooltips, etc)
    maxDarkLightNesting: 2
  },
  defaultConfig = {
    animations,
    media,
    shorthands,
    themes: defaultThemes,
    tokens,
    fonts,
    selectionStyles,
    settings
  };
export { animations2 as animations, breakpoints, createSystemFont, createThemes, defaultConfig, fonts2 as fonts, media2 as media, mediaQueryDefaultActive2 as mediaQueryDefaultActive, selectionStyles, settings, shorthands2 as shorthands, defaultThemes2 as themes, tokens2 as tokens };
