// @tamagui/colors@1.144.4/dist/cjs/legacy.cjs
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
    for (var name in all) __defProp(target, name, {
      get: all[name],
      enumerable: !0
    });
  },
  __copyProps = (to, from, except, desc) => {
    if (from && typeof from == "object" || typeof from == "function") for (let key of __getOwnPropNames(from)) !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, {
      get: () => from[key],
      enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable
    });
    return to;
  };
var __toCommonJS = mod => __copyProps(__defProp({}, "__esModule", {
  value: !0
}), mod);
var legacy_exports = {};
__export(legacy_exports, {
  amber: () => import_amber2.amber,
  amberA: () => import_amberA2.amberA,
  amberDark: () => import_amber.amber,
  amberDarkA: () => import_amberA.amberA,
  blackA: () => import_blackA.blackA,
  blue: () => import_blue2.blue,
  blueA: () => import_blueA2.blueA,
  blueDark: () => import_blue.blue,
  blueDarkA: () => import_blueA.blueA,
  bronze: () => import_bronze2.bronze,
  bronzeA: () => import_bronzeA2.bronzeA,
  bronzeDark: () => import_bronze.bronze,
  bronzeDarkA: () => import_bronzeA.bronzeA,
  brown: () => import_brown2.brown,
  brownA: () => import_brownA2.brownA,
  brownDark: () => import_brown.brown,
  brownDarkA: () => import_brownA.brownA,
  crimson: () => import_crimson2.crimson,
  crimsonA: () => import_crimsonA2.crimsonA,
  crimsonDark: () => import_crimson.crimson,
  crimsonDarkA: () => import_crimsonA.crimsonA,
  cyan: () => import_cyan2.cyan,
  cyanA: () => import_cyanA2.cyanA,
  cyanDark: () => import_cyan.cyan,
  cyanDarkA: () => import_cyanA.cyanA,
  gold: () => import_gold2.gold,
  goldA: () => import_goldA2.goldA,
  goldDark: () => import_gold.gold,
  goldDarkA: () => import_goldA.goldA,
  grass: () => import_grass2.grass,
  grassA: () => import_grassA2.grassA,
  grassDark: () => import_grass.grass,
  grassDarkA: () => import_grassA.grassA,
  gray: () => import_gray2.gray,
  grayA: () => import_grayA2.grayA,
  grayDark: () => import_gray.gray,
  grayDarkA: () => import_grayA.grayA,
  green: () => import_green2.green,
  greenA: () => import_greenA2.greenA,
  greenDark: () => import_green.green,
  greenDarkA: () => import_greenA.greenA,
  indigo: () => import_indigo2.indigo,
  indigoA: () => import_indigoA2.indigoA,
  indigoDark: () => import_indigo.indigo,
  indigoDarkA: () => import_indigoA.indigoA,
  lime: () => import_lime2.lime,
  limeA: () => import_limeA2.limeA,
  limeDark: () => import_lime.lime,
  limeDarkA: () => import_limeA.limeA,
  mauve: () => import_mauve2.mauve,
  mauveA: () => import_mauveA2.mauveA,
  mauveDark: () => import_mauve.mauve,
  mauveDarkA: () => import_mauveA.mauveA,
  mint: () => import_mint2.mint,
  mintA: () => import_mintA2.mintA,
  mintDark: () => import_mint.mint,
  mintDarkA: () => import_mintA.mintA,
  olive: () => import_olive2.olive,
  oliveA: () => import_oliveA2.oliveA,
  oliveDark: () => import_olive.olive,
  oliveDarkA: () => import_oliveA.oliveA,
  orange: () => import_orange2.orange,
  orangeA: () => import_orangeA2.orangeA,
  orangeDark: () => import_orange.orange,
  orangeDarkA: () => import_orangeA.orangeA,
  pink: () => import_pink2.pink,
  pinkA: () => import_pinkA2.pinkA,
  pinkDark: () => import_pink.pink,
  pinkDarkA: () => import_pinkA.pinkA,
  plum: () => import_plum2.plum,
  plumA: () => import_plumA2.plumA,
  plumDark: () => import_plum.plum,
  plumDarkA: () => import_plumA.plumA,
  purple: () => import_purple2.purple,
  purpleA: () => import_purpleA2.purpleA,
  purpleDark: () => import_purple.purple,
  purpleDarkA: () => import_purpleA.purpleA,
  red: () => import_red2.red,
  redA: () => import_redA2.redA,
  redDark: () => import_red.red,
  redDarkA: () => import_redA.redA,
  sage: () => import_sage2.sage,
  sageA: () => import_sageA2.sageA,
  sageDark: () => import_sage.sage,
  sageDarkA: () => import_sageA.sageA,
  sand: () => import_sand2.sand,
  sandA: () => import_sandA2.sandA,
  sandDark: () => import_sand.sand,
  sandDarkA: () => import_sandA.sandA,
  sky: () => import_sky2.sky,
  skyA: () => import_skyA2.skyA,
  skyDark: () => import_sky.sky,
  skyDarkA: () => import_skyA.skyA,
  slate: () => import_slate2.slate,
  slateA: () => import_slateA2.slateA,
  slateDark: () => import_slate.slate,
  slateDarkA: () => import_slateA.slateA,
  teal: () => import_teal2.teal,
  tealA: () => import_tealA2.tealA,
  tealDark: () => import_teal.teal,
  tealDarkA: () => import_tealA.tealA,
  tomato: () => import_tomato2.tomato,
  tomatoA: () => import_tomatoA2.tomatoA,
  tomatoDark: () => import_tomato.tomato,
  tomatoDarkA: () => import_tomatoA.tomatoA,
  violet: () => import_violet2.violet,
  violetA: () => import_violetA2.violetA,
  violetDark: () => import_violet.violet,
  violetDarkA: () => import_violetA.violetA,
  whiteA: () => import_whiteA.whiteA,
  yellow: () => import_yellow2.yellow,
  yellowA: () => import_yellowA2.yellowA,
  yellowDark: () => import_yellow.yellow,
  yellowDarkA: () => import_yellowA.yellowA
});
module.exports = __toCommonJS(legacy_exports);
var import_amber = require("./legacy/dark/amber.cjs"),
  import_blue = require("./legacy/dark/blue.cjs"),
  import_bronze = require("./legacy/dark/bronze.cjs"),
  import_brown = require("./legacy/dark/brown.cjs"),
  import_crimson = require("./legacy/dark/crimson.cjs"),
  import_cyan = require("./legacy/dark/cyan.cjs"),
  import_gold = require("./legacy/dark/gold.cjs"),
  import_grass = require("./legacy/dark/grass.cjs"),
  import_gray = require("./legacy/dark/gray.cjs"),
  import_green = require("./legacy/dark/green.cjs"),
  import_indigo = require("./legacy/dark/indigo.cjs"),
  import_lime = require("./legacy/dark/lime.cjs"),
  import_mauve = require("./legacy/dark/mauve.cjs"),
  import_mint = require("./legacy/dark/mint.cjs"),
  import_olive = require("./legacy/dark/olive.cjs"),
  import_orange = require("./legacy/dark/orange.cjs"),
  import_pink = require("./legacy/dark/pink.cjs"),
  import_plum = require("./legacy/dark/plum.cjs"),
  import_purple = require("./legacy/dark/purple.cjs"),
  import_red = require("./legacy/dark/red.cjs"),
  import_sage = require("./legacy/dark/sage.cjs"),
  import_sand = require("./legacy/dark/sand.cjs"),
  import_sky = require("./legacy/dark/sky.cjs"),
  import_slate = require("./legacy/dark/slate.cjs"),
  import_teal = require("./legacy/dark/teal.cjs"),
  import_tomato = require("./legacy/dark/tomato.cjs"),
  import_violet = require("./legacy/dark/violet.cjs"),
  import_yellow = require("./legacy/dark/yellow.cjs"),
  import_amberA = require("./legacy/dark/amberA.cjs"),
  import_blueA = require("./legacy/dark/blueA.cjs"),
  import_bronzeA = require("./legacy/dark/bronzeA.cjs"),
  import_brownA = require("./legacy/dark/brownA.cjs"),
  import_crimsonA = require("./legacy/dark/crimsonA.cjs"),
  import_cyanA = require("./legacy/dark/cyanA.cjs"),
  import_goldA = require("./legacy/dark/goldA.cjs"),
  import_grassA = require("./legacy/dark/grassA.cjs"),
  import_grayA = require("./legacy/dark/grayA.cjs"),
  import_greenA = require("./legacy/dark/greenA.cjs"),
  import_indigoA = require("./legacy/dark/indigoA.cjs"),
  import_limeA = require("./legacy/dark/limeA.cjs"),
  import_mauveA = require("./legacy/dark/mauveA.cjs"),
  import_mintA = require("./legacy/dark/mintA.cjs"),
  import_oliveA = require("./legacy/dark/oliveA.cjs"),
  import_orangeA = require("./legacy/dark/orangeA.cjs"),
  import_pinkA = require("./legacy/dark/pinkA.cjs"),
  import_plumA = require("./legacy/dark/plumA.cjs"),
  import_purpleA = require("./legacy/dark/purpleA.cjs"),
  import_redA = require("./legacy/dark/redA.cjs"),
  import_sageA = require("./legacy/dark/sageA.cjs"),
  import_sandA = require("./legacy/dark/sandA.cjs"),
  import_skyA = require("./legacy/dark/skyA.cjs"),
  import_slateA = require("./legacy/dark/slateA.cjs"),
  import_tealA = require("./legacy/dark/tealA.cjs"),
  import_tomatoA = require("./legacy/dark/tomatoA.cjs"),
  import_violetA = require("./legacy/dark/violetA.cjs"),
  import_yellowA = require("./legacy/dark/yellowA.cjs"),
  import_amber2 = require("./legacy/light/amber.cjs"),
  import_blue2 = require("./legacy/light/blue.cjs"),
  import_bronze2 = require("./legacy/light/bronze.cjs"),
  import_brown2 = require("./legacy/light/brown.cjs"),
  import_crimson2 = require("./legacy/light/crimson.cjs"),
  import_cyan2 = require("./legacy/light/cyan.cjs"),
  import_gold2 = require("./legacy/light/gold.cjs"),
  import_grass2 = require("./legacy/light/grass.cjs"),
  import_gray2 = require("./legacy/light/gray.cjs"),
  import_green2 = require("./legacy/light/green.cjs"),
  import_indigo2 = require("./legacy/light/indigo.cjs"),
  import_lime2 = require("./legacy/light/lime.cjs"),
  import_mauve2 = require("./legacy/light/mauve.cjs"),
  import_mint2 = require("./legacy/light/mint.cjs"),
  import_olive2 = require("./legacy/light/olive.cjs"),
  import_orange2 = require("./legacy/light/orange.cjs"),
  import_pink2 = require("./legacy/light/pink.cjs"),
  import_plum2 = require("./legacy/light/plum.cjs"),
  import_purple2 = require("./legacy/light/purple.cjs"),
  import_red2 = require("./legacy/light/red.cjs"),
  import_sage2 = require("./legacy/light/sage.cjs"),
  import_sand2 = require("./legacy/light/sand.cjs"),
  import_sky2 = require("./legacy/light/sky.cjs"),
  import_slate2 = require("./legacy/light/slate.cjs"),
  import_teal2 = require("./legacy/light/teal.cjs"),
  import_tomato2 = require("./legacy/light/tomato.cjs"),
  import_violet2 = require("./legacy/light/violet.cjs"),
  import_yellow2 = require("./legacy/light/yellow.cjs"),
  import_amberA2 = require("./legacy/light/amberA.cjs"),
  import_blueA2 = require("./legacy/light/blueA.cjs"),
  import_bronzeA2 = require("./legacy/light/bronzeA.cjs"),
  import_brownA2 = require("./legacy/light/brownA.cjs"),
  import_crimsonA2 = require("./legacy/light/crimsonA.cjs"),
  import_cyanA2 = require("./legacy/light/cyanA.cjs"),
  import_goldA2 = require("./legacy/light/goldA.cjs"),
  import_grassA2 = require("./legacy/light/grassA.cjs"),
  import_grayA2 = require("./legacy/light/grayA.cjs"),
  import_greenA2 = require("./legacy/light/greenA.cjs"),
  import_indigoA2 = require("./legacy/light/indigoA.cjs"),
  import_limeA2 = require("./legacy/light/limeA.cjs"),
  import_mauveA2 = require("./legacy/light/mauveA.cjs"),
  import_mintA2 = require("./legacy/light/mintA.cjs"),
  import_oliveA2 = require("./legacy/light/oliveA.cjs"),
  import_orangeA2 = require("./legacy/light/orangeA.cjs"),
  import_pinkA2 = require("./legacy/light/pinkA.cjs"),
  import_plumA2 = require("./legacy/light/plumA.cjs"),
  import_purpleA2 = require("./legacy/light/purpleA.cjs"),
  import_redA2 = require("./legacy/light/redA.cjs"),
  import_sageA2 = require("./legacy/light/sageA.cjs"),
  import_sandA2 = require("./legacy/light/sandA.cjs"),
  import_skyA2 = require("./legacy/light/skyA.cjs"),
  import_slateA2 = require("./legacy/light/slateA.cjs"),
  import_tealA2 = require("./legacy/light/tealA.cjs"),
  import_tomatoA2 = require("./legacy/light/tomatoA.cjs"),
  import_violetA2 = require("./legacy/light/violetA.cjs"),
  import_yellowA2 = require("./legacy/light/yellowA.cjs"),
  import_blackA = require("./blackA.cjs"),
  import_whiteA = require("./whiteA.cjs");
