// @tamagui/helpers@1.144.4/dist/esm/webOnlyStyleProps.native.js
var webOnlyStylePropsView = {
    transition: !0,
    textWrap: !0,
    backdropFilter: !0,
    WebkitBackdropFilter: !0,
    // These background-* props are web-only CSS, but NOT backgroundColor
    background: !0,
    // CSS border shorthand (e.g., "1px solid red") - RN only supports individual props
    border: !0,
    borderTop: !0,
    borderRight: !0,
    borderBottom: !0,
    borderLeft: !0,
    backgroundAttachment: !0,
    backgroundBlendMode: !0,
    backgroundClip: !0,
    backgroundImage: !0,
    backgroundOrigin: !0,
    backgroundPosition: !0,
    backgroundRepeat: !0,
    backgroundSize: !0,
    borderBottomStyle: !0,
    borderImage: !0,
    borderLeftStyle: !0,
    borderRightStyle: !0,
    borderTopStyle: !0,
    // boxSizing: now supported in RN 0.77+ (New Architecture)
    caretColor: !0,
    clipPath: !0,
    contain: !0,
    containerType: !0,
    content: !0,
    // cursor: now cross-platform - supported on iOS 17+ (trackpad/stylus) with limited values
    float: !0,
    mask: !0,
    maskBorder: !0,
    maskBorderMode: !0,
    maskBorderOutset: !0,
    maskBorderRepeat: !0,
    maskBorderSlice: !0,
    maskBorderSource: !0,
    maskBorderWidth: !0,
    maskClip: !0,
    maskComposite: !0,
    maskImage: !0,
    maskMode: !0,
    maskOrigin: !0,
    maskPosition: !0,
    maskRepeat: !0,
    maskSize: !0,
    maskType: !0,
    // mixBlendMode: now supported in RN 0.77+ (New Architecture)
    objectFit: !0,
    objectPosition: !0,
    // outline*: now supported in RN 0.77+ (New Architecture)
    overflowBlock: !0,
    overflowInline: !0,
    overflowX: !0,
    overflowY: !0,
    // NOTE: pointerEvents is NOT web-only - it's a core React Native View prop (not a style)
    pointerEvents: !0,
    scrollbarWidth: !0,
    textEmphasis: !0,
    touchAction: !0,
    transformStyle: !0,
    userSelect: !0,
    willChange: !0
  },
  webOnlyStylePropsText = {
    whiteSpace: !0,
    wordWrap: !0,
    textOverflow: !0,
    textDecorationDistance: !0,
    // cursor: now cross-platform - in stylePropsView
    WebkitLineClamp: !0,
    WebkitBoxOrient: !0
  };
export { webOnlyStylePropsText, webOnlyStylePropsView };
