// @tamagui/themes@1.144.4/dist/esm/defaultColorThemes.js
import {
  blue,
  blueDark,
  gray,
  grayDark,
  green,
  greenDark,
  red,
  redDark,
  yellow,
  yellowDark
} from "@tamagui/colors";
import { darkPalette, lightPalette, neutralPalette } from "./v5-palettes";
function paletteToNamedColors(name, palette) {
  return {
    [`${name}1`]: palette[0],
    [`${name}2`]: palette[1],
    [`${name}3`]: palette[2],
    [`${name}4`]: palette[3],
    [`${name}5`]: palette[4],
    [`${name}6`]: palette[5],
    [`${name}7`]: palette[6],
    [`${name}8`]: palette[7],
    [`${name}9`]: palette[8],
    [`${name}10`]: palette[9],
    [`${name}11`]: palette[10],
    [`${name}12`]: palette[11]
  };
}
const defaultColorThemes = {
  black: {
    palette: {
      dark: Object.values(darkPalette),
      light: Object.values(darkPalette)
    }
  },
  white: {
    palette: {
      dark: Object.values(lightPalette),
      light: Object.values(lightPalette)
    }
  },
  gray: {
    palette: {
      dark: Object.values(grayDark),
      light: Object.values(gray)
    }
  },
  blue: {
    palette: {
      dark: Object.values(blueDark),
      light: Object.values(blue)
    }
  },
  red: {
    palette: {
      dark: Object.values(redDark),
      light: Object.values(red)
    }
  },
  yellow: {
    palette: {
      dark: Object.values(yellowDark),
      light: Object.values(yellow)
    }
  },
  green: {
    palette: {
      dark: Object.values(greenDark),
      light: Object.values(green)
    }
  },
  neutral: {
    palette: {
      dark: neutralPalette,
      light: neutralPalette
    }
  }
};
function colorThemesToExtra(colorThemes, scheme) {
  const extra = {};
  for (const [name, theme] of Object.entries(colorThemes)) {
    const palette = scheme === "light" ? theme.palette.light : theme.palette.dark, namedColors = paletteToNamedColors(name, palette);
    Object.assign(extra, namedColors);
  }
  return extra;
}
export {
  colorThemesToExtra,
  defaultColorThemes
};
