// @tamagui/web@1.144.4/dist/cjs/config.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: !0 });
}, __copyProps = (to, from, except, desc) => {
  if (from && typeof from == "object" || typeof from == "function")
    for (let key of __getOwnPropNames(from))
      !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: !0 }), mod);
var config_exports = {};
__export(config_exports, {
  configListeners: () => configListeners,
  devConfig: () => devConfig,
  getConfig: () => getConfig,
  getConfigMaybe: () => getConfigMaybe,
  getFont: () => getFont,
  getSetting: () => getSetting,
  getThemes: () => getThemes,
  getToken: () => getToken,
  getTokenObject: () => getTokenObject,
  getTokenValue: () => getTokenValue,
  getTokens: () => getTokens,
  onConfiguredOnce: () => onConfiguredOnce,
  setConfig: () => setConfig,
  setConfigFont: () => setConfigFont,
  setTokens: () => setTokens,
  setupDev: () => setupDev,
  updateConfig: () => updateConfig,
  useTokens: () => useTokens
});
module.exports = __toCommonJS(config_exports);
var import_constants = require("@tamagui/constants"), import_constants2 = require("./constants/constants");
let conf, setConfigCalledByThisInstance = !1;
const haventCalledErrorMessage = process.env.NODE_ENV === "development" ? `
Haven't called createTamagui yet. ${import_constants2.MISSING_THEME_MESSAGE}
` : "\u274C Error 001", getConfigFromGlobalOrLocal = () => conf || (globalThis.__tamaguiConfig ? (process.env.NODE_ENV === "development" && !globalThis.__tamaguiHasWarnedGlobalFallback && !globalThis.__tamaguiPendingCheck && (globalThis.__tamaguiPendingCheck = !0, setTimeout(() => {
  !setConfigCalledByThisInstance && !globalThis.__tamaguiHasWarnedGlobalFallback && (globalThis.__tamaguiHasWarnedGlobalFallback = !0, console.warn(
    "Tamagui: Using global config fallback. This may indicate duplicate tamagui instances (e.g., from Vite SSR bundling). This is handled automatically, but may cause issues."
  )), globalThis.__tamaguiPendingCheck = !1;
}, 500)), globalThis.__tamaguiConfig) : null), getSetting = (key) => {
  const config = getConfigFromGlobalOrLocal();
  if (process.env.NODE_ENV === "development" && !config)
    throw new Error(haventCalledErrorMessage);
  return config.settings[key] ?? // @ts-expect-error
  config[key];
}, setConfig = (next) => {
  setConfigCalledByThisInstance = !0, conf = next, globalThis.__tamaguiConfig = next;
}, setConfigFont = (name, font, fontParsed) => {
  const config = getConfigFromGlobalOrLocal();
  if (process.env.NODE_ENV === "development" && !config)
    throw new Error(haventCalledErrorMessage);
  config.fonts[name] = font, config.fontsParsed[`$${name}`] = fontParsed;
}, getConfig = () => {
  const config = getConfigFromGlobalOrLocal();
  if (!config)
    throw new Error(
      process.env.NODE_ENV !== "production" ? "Missing tamagui config, you either have a duplicate config, or haven't set it up. Be sure createTamagui is called before rendering. Also, make sure all of your tamagui dependencies are on the same version (`tamagui`, `@tamagui/package-name`, etc.) not just in your package.json, but in your lockfile." : "Err0"
    );
  return config;
}, getConfigMaybe = () => getConfigFromGlobalOrLocal();
let tokensMerged;
function setTokens(_) {
  tokensMerged = _;
}
const getTokens = ({
  prefixed
} = {}) => {
  const config = getConfigFromGlobalOrLocal();
  if (process.env.NODE_ENV === "development" && !config)
    throw new Error(haventCalledErrorMessage);
  const { tokens, tokensParsed } = config;
  return prefixed === !1 ? tokens : prefixed === !0 ? tokensParsed : tokensMerged;
}, getTokenObject = (value, group) => getConfigFromGlobalOrLocal().specificTokens[value] ?? (group ? tokensMerged[group]?.[value] : tokensMerged[Object.keys(tokensMerged).find((cat) => tokensMerged[cat][value]) || ""]?.[value]), getToken = (value, group, useVariable = import_constants.isWeb) => {
  const token = getTokenObject(value, group);
  return useVariable ? token?.variable : token?.val;
}, getTokenValue = (value, group) => {
  if (!(value === "unset" || value === "auto"))
    return getToken(value, group, !1);
}, useTokens = getTokens, getThemes = () => getConfigFromGlobalOrLocal().themes, configListeners = /* @__PURE__ */ new Set(), onConfiguredOnce = (cb) => {
  const config = getConfigFromGlobalOrLocal();
  config ? cb(config) : configListeners.add(cb);
}, updateConfig = (key, value) => {
  const config = getConfigFromGlobalOrLocal();
  Object.assign(config[key], value);
}, getFont = (name) => {
  const conf2 = getConfig();
  return conf2.fontsParsed[name] ?? Object.entries(conf2.fontsParsed).find(
    ([k]) => conf2.fontsParsed[k]?.family?.val === name
  )?.[1];
};
let devConfig;
function setupDev(conf2) {
  process.env.NODE_ENV === "development" && (devConfig = conf2);
}
