// @tamagui/web@1.144.4/dist/esm/helpers/createDesignSystem.native.js
import { isWeb } from "@tamagui/constants";
import { getVariableVariable, isVariable } from "../createVariable.native.js";
import { registerCSSVariable, variableToCSS } from "./registerCSSVariable.native.js";
import { getThemeCSSRules } from "./getThemeCSSRules.native.js";
import "./insertStyleRule.native.js";
function getFontPropertyDeclarations(fontParsed) {
  var tokenKey = arguments.length > 1 && arguments[1] !== void 0 ? arguments[1] : "$true",
    props = ["font-family: var(--f-family)"],
    getVarRef = function (obj) {
      var val = obj?.[tokenKey];
      if (isVariable(val)) return getVariableVariable(val);
    },
    letterSpacing = getVarRef(fontParsed.letterSpacing);
  letterSpacing && props.push(`letter-spacing: ${letterSpacing}`);
  var lineHeight = getVarRef(fontParsed.lineHeight);
  lineHeight && props.push(`line-height: ${lineHeight}`);
  var fontStyle = getVarRef(fontParsed.style);
  fontStyle && props.push(`font-style: ${fontStyle}`);
  var fontWeight = getVarRef(fontParsed.weight);
  return fontWeight && props.push(`font-weight: ${fontWeight}`), props;
}
function createTokenCSS(tokens, shouldTokenCategoryHaveUnits) {
  if (!process.env.TAMAGUI_DID_OUTPUT_CSS) {
    var declarations = [],
      sortedTokenKeys = Object.keys(tokens).sort(),
      _iteratorNormalCompletion = !0,
      _didIteratorError = !1,
      _iteratorError = void 0;
    try {
      for (var _iterator = sortedTokenKeys[Symbol.iterator](), _step; !(_iteratorNormalCompletion = (_step = _iterator.next()).done); _iteratorNormalCompletion = !0) {
        var key = _step.value,
          sortedSubKeys = Object.keys(tokens[key]).sort(),
          _iteratorNormalCompletion1 = !0,
          _didIteratorError1 = !1,
          _iteratorError1 = void 0;
        try {
          for (var _iterator1 = sortedSubKeys[Symbol.iterator](), _step1; !(_iteratorNormalCompletion1 = (_step1 = _iterator1.next()).done); _iteratorNormalCompletion1 = !0) {
            var skey = _step1.value,
              variable = tokens[key][skey];
            if (isWeb) {
              registerCSSVariable(variable);
              var variableNeedsPx = variable.needsPx === !0,
                categoryNeedsPx = shouldTokenCategoryHaveUnits(key),
                shouldBeUnitless = !(variableNeedsPx || categoryNeedsPx);
              declarations.push(variableToCSS(variable, shouldBeUnitless));
            }
          }
        } catch (err) {
          _didIteratorError1 = !0, _iteratorError1 = err;
        } finally {
          try {
            !_iteratorNormalCompletion1 && _iterator1.return != null && _iterator1.return();
          } finally {
            if (_didIteratorError1) throw _iteratorError1;
          }
        }
      }
    } catch (err) {
      _didIteratorError = !0, _iteratorError = err;
    } finally {
      try {
        !_iteratorNormalCompletion && _iterator.return != null && _iterator.return();
      } finally {
        if (_didIteratorError) throw _iteratorError;
      }
    }
    return declarations;
  }
  return [];
}
function createFontCSS(fontsParsed, registerFontVariables) {
  if (!process.env.TAMAGUI_DID_OUTPUT_CSS) {
    var fontDeclarations = {};
    if (!fontsParsed) return fontDeclarations;
    var sortedFontKeys = Object.keys(fontsParsed).sort(),
      _iteratorNormalCompletion = !0,
      _didIteratorError = !1,
      _iteratorError = void 0;
    try {
      for (var _iterator = sortedFontKeys[Symbol.iterator](), _step; !(_iteratorNormalCompletion = (_step = _iterator.next()).done); _iteratorNormalCompletion = !0) {
        var key = _step.value,
          fontParsed = fontsParsed[key],
          [name, language] = key.includes("_") ? key.split("_") : [key],
          fontVars = registerFontVariables(fontParsed);
        fontDeclarations[key] = {
          name: name.slice(1),
          declarations: fontVars,
          language,
          fontParsed
        };
      }
    } catch (err) {
      _didIteratorError = !0, _iteratorError = err;
    } finally {
      try {
        !_iteratorNormalCompletion && _iterator.return != null && _iterator.return();
      } finally {
        if (_didIteratorError) throw _iteratorError;
      }
    }
    return fontDeclarations;
  }
  return {};
}
function buildCSSRuleSets(declarations, fontDeclarations) {
  var defaultFontToken = arguments.length > 2 && arguments[2] !== void 0 ? arguments[2] : "$true";
  if (!process.env.TAMAGUI_DID_OUTPUT_CSS) {
    let declarationsToRuleSet2 = function (decs) {
      var selector = arguments.length > 1 && arguments[1] !== void 0 ? arguments[1] : "";
      return `:root${selector} {${sep}${[...decs].join(`;${sep}`)}
}`;
    };
    var declarationsToRuleSet = declarationsToRuleSet2,
      cssRuleSets = [],
      sep = " ";
    declarations.length && cssRuleSets.push(declarationsToRuleSet2(declarations));
    var fontSelectors = [],
      sortedFontDeclarationKeys = Object.keys(fontDeclarations).sort(),
      _iteratorNormalCompletion = !0,
      _didIteratorError = !1,
      _iteratorError = void 0;
    try {
      for (var _iterator = sortedFontDeclarationKeys[Symbol.iterator](), _step; !(_iteratorNormalCompletion = (_step = _iterator.next()).done); _iteratorNormalCompletion = !0) {
        var key = _step.value,
          {
            name,
            declarations: _$declarations,
            language = "default"
          } = fontDeclarations[key],
          fontSelector = `.font_${name}`;
        fontSelectors.push(fontSelector);
        var langSelector = `:root .t_lang-${name}-${language} ${fontSelector}`,
          selectors = language === "default" ? ` ${fontSelector}, ${langSelector}` : langSelector,
          specificRuleSet = declarationsToRuleSet2(_$declarations, selectors);
        cssRuleSets.push(specificRuleSet);
      }
    } catch (err) {
      _didIteratorError = !0, _iteratorError = err;
    } finally {
      try {
        !_iteratorNormalCompletion && _iterator.return != null && _iterator.return();
      } finally {
        if (_didIteratorError) throw _iteratorError;
      }
    }
    if (fontSelectors.length) {
      var firstFont = fontDeclarations[sortedFontDeclarationKeys[0]];
      if (firstFont?.fontParsed) {
        var fontProps = getFontPropertyDeclarations(firstFont.fontParsed, defaultFontToken),
          sharedSelectors = [...fontSelectors, ".is_View"].join(", ");
        cssRuleSets.push(`${sharedSelectors} {${fontProps.join("; ")}}`);
      }
    }
    return cssRuleSets;
  }
  return [];
}
function createThemeCSS(dedupedThemes, configIn) {
  if (!process.env.TAMAGUI_DID_OUTPUT_CSS) {
    var themeRuleSets = [];
    if (isWeb) {
      var _iteratorNormalCompletion = !0,
        _didIteratorError = !1,
        _iteratorError = void 0;
      try {
        for (var _iterator = dedupedThemes[Symbol.iterator](), _step; !(_iteratorNormalCompletion = (_step = _iterator.next()).done); _iteratorNormalCompletion = !0) {
          var {
              names,
              theme
            } = _step.value,
            nextRules = getThemeCSSRules({
              config: configIn,
              themeName: names[0],
              names,
              theme
            });
          themeRuleSets = [...themeRuleSets, ...nextRules];
        }
      } catch (err) {
        _didIteratorError = !0, _iteratorError = err;
      } finally {
        try {
          !_iteratorNormalCompletion && _iterator.return != null && _iterator.return();
        } finally {
          if (_didIteratorError) throw _iteratorError;
        }
      }
    }
    return themeRuleSets;
  }
  return [];
}
function getCSS(themeConfig) {
  var opts = arguments.length > 1 && arguments[1] !== void 0 ? arguments[1] : {},
    lastIndex = arguments.length > 2 ? arguments[2] : void 0;
  if (!process.env.TAMAGUI_DID_OUTPUT_CSS && !1) {
    var separator, sinceLastCall, exclude;
    if (sinceLastCall && lastIndex.value >= 0) var rules, newRules;
    var runtimeStyles, themeRules, autoVarCSS, designSystem;
  }
  return "";
}
export { buildCSSRuleSets, createFontCSS, createThemeCSS, createTokenCSS, getCSS, getFontPropertyDeclarations };
