// @tamagui/web@1.144.4/dist/esm/helpers/createDesignSystem.mjs
import { isWeb } from "@tamagui/constants";
import { getVariableVariable, isVariable } from "../createVariable.mjs";
import { autoVariables, registerCSSVariable, variableToCSS } from "./registerCSSVariable.mjs";
import { getThemeCSSRules } from "./getThemeCSSRules.mjs";
import { getAllRules } from "./insertStyleRule.mjs";
function getFontPropertyDeclarations(fontParsed, tokenKey = "$true") {
  const props = ["font-family: var(--f-family)"],
    getVarRef = obj => {
      const val = obj?.[tokenKey];
      if (isVariable(val)) return getVariableVariable(val);
    },
    letterSpacing = getVarRef(fontParsed.letterSpacing);
  letterSpacing && props.push(`letter-spacing: ${letterSpacing}`);
  const lineHeight = getVarRef(fontParsed.lineHeight);
  lineHeight && props.push(`line-height: ${lineHeight}`);
  const fontStyle = getVarRef(fontParsed.style);
  fontStyle && props.push(`font-style: ${fontStyle}`);
  const fontWeight = getVarRef(fontParsed.weight);
  return fontWeight && props.push(`font-weight: ${fontWeight}`), props;
}
function createTokenCSS(tokens, shouldTokenCategoryHaveUnits) {
  if (!process.env.TAMAGUI_DID_OUTPUT_CSS) {
    const declarations = [],
      sortedTokenKeys = Object.keys(tokens).sort();
    for (const key of sortedTokenKeys) {
      const sortedSubKeys = Object.keys(tokens[key]).sort();
      for (const skey of sortedSubKeys) {
        const variable = tokens[key][skey];
        if (isWeb) {
          registerCSSVariable(variable);
          const variableNeedsPx = variable.needsPx === !0,
            categoryNeedsPx = shouldTokenCategoryHaveUnits(key),
            shouldBeUnitless = !(variableNeedsPx || categoryNeedsPx);
          declarations.push(variableToCSS(variable, shouldBeUnitless));
        }
      }
    }
    return declarations;
  }
  return [];
}
function createFontCSS(fontsParsed, registerFontVariables) {
  if (!process.env.TAMAGUI_DID_OUTPUT_CSS) {
    const fontDeclarations = {};
    if (!fontsParsed) return fontDeclarations;
    const sortedFontKeys = Object.keys(fontsParsed).sort();
    for (const key of sortedFontKeys) {
      const fontParsed = fontsParsed[key],
        [name, language] = key.includes("_") ? key.split("_") : [key],
        fontVars = registerFontVariables(fontParsed);
      fontDeclarations[key] = {
        name: name.slice(1),
        declarations: fontVars,
        language,
        fontParsed
      };
    }
    return fontDeclarations;
  }
  return {};
}
function buildCSSRuleSets(declarations, fontDeclarations, defaultFontToken = "$true") {
  if (!process.env.TAMAGUI_DID_OUTPUT_CSS) {
    let declarationsToRuleSet = function (decs, selector = "") {
      return `:root${selector} { ${[...decs].join("; ")}
}`;
    };
    const cssRuleSets = [];
    declarations.length && cssRuleSets.push(declarationsToRuleSet(declarations));
    const fontSelectors = [],
      sortedFontDeclarationKeys = Object.keys(fontDeclarations).sort();
    for (const key of sortedFontDeclarationKeys) {
      const {
          name,
          declarations: declarations2,
          language = "default"
        } = fontDeclarations[key],
        fontSelector = `.font_${name}`;
      fontSelectors.push(fontSelector);
      const langSelector = `:root .t_lang-${name}-${language} ${fontSelector}`,
        selectors = language === "default" ? ` ${fontSelector}, ${langSelector}` : langSelector,
        specificRuleSet = declarationsToRuleSet(declarations2, selectors);
      cssRuleSets.push(specificRuleSet);
    }
    if (fontSelectors.length) {
      const firstFont = fontDeclarations[sortedFontDeclarationKeys[0]];
      if (firstFont?.fontParsed) {
        const fontProps = getFontPropertyDeclarations(firstFont.fontParsed, defaultFontToken),
          sharedSelectors = [...fontSelectors, ".is_View"].join(", ");
        cssRuleSets.push(`${sharedSelectors} {${fontProps.join("; ")}}`);
      }
    }
    return cssRuleSets;
  }
  return [];
}
function createThemeCSS(dedupedThemes, configIn) {
  if (!process.env.TAMAGUI_DID_OUTPUT_CSS) {
    let themeRuleSets = [];
    if (isWeb) for (const {
      names,
      theme
    } of dedupedThemes) {
      const nextRules = getThemeCSSRules({
        config: configIn,
        themeName: names[0],
        names,
        theme
      });
      themeRuleSets = [...themeRuleSets, ...nextRules];
    }
    return themeRuleSets;
  }
  return [];
}
function getCSS(themeConfig, opts = {}, lastIndex) {
  if (!process.env.TAMAGUI_DID_OUTPUT_CSS) {
    const {
      separator = `
`,
      sinceLastCall,
      exclude
    } = opts;
    if (sinceLastCall && lastIndex.value >= 0) {
      const rules = getAllRules(),
        newRules = rules.slice(lastIndex.value);
      return lastIndex.value = rules.length, newRules.join(separator);
    }
    lastIndex.value = 0;
    const runtimeStyles = getAllRules().join(separator);
    if (exclude === "design-system") return runtimeStyles;
    const themeRules = exclude ? "" : themeConfig.getThemeRulesSets().join(separator);
    return `${`._ovs-contain {overscroll-behavior:contain;}
.is_View { display: flex; align-items: stretch; flex-direction: column; flex-basis: auto; box-sizing: border-box; min-height: 0; min-width: 0; flex-shrink: 0; }
.is_Text { display: inline; box-sizing: border-box; word-wrap: break-word; white-space: pre-wrap; margin: 0; }
@scope (.is_Text) to (.is_View) { .is_Text { white-space: inherit; word-wrap: inherit; } }
._dsp_contents {display:contents;}
._no_backdrop::backdrop {display: none;}
.is_Input::selection, .is_TextArea::selection {background-color: var(--selectionColor);}
.is_Input::placeholder, .is_TextArea::placeholder {color: var(--placeholderColor);}
._hsb-x > div::-webkit-scrollbar:horizontal { display: none; }
._hsb-y > div::-webkit-scrollbar:vertical { display: none; }
${autoVariables.length ? `:root{${autoVariables.map(v => `--${v.name}:${v.val}`).join(";")}}` : ""}
${themeConfig.cssRuleSets.join(separator)}`}
${themeRules}
${runtimeStyles}`;
  }
  return "";
}
export { buildCSSRuleSets, createFontCSS, createThemeCSS, createTokenCSS, getCSS, getFontPropertyDeclarations };
