// @tamagui/web@1.144.4/dist/esm/helpers/parseNativeStyle.js
function parseNativeStyle(key, cssString, tokenMap) {
  switch (key) {
    case "backgroundImage":
      return parseBackgroundImage(cssString, tokenMap);
    case "boxShadow":
      return parseBoxShadow(cssString, tokenMap);
    case "textShadow":
      return parseTextShadow(cssString, tokenMap);
    default:
      return;
  }
}
function resolveColor(raw, tokenMap) {
  return tokenMap && tokenMap.has(raw) ? tokenMap.get(raw) : raw;
}
function parseBackgroundImage(css, tokenMap) {
  const match = css.match(/^linear-gradient\((.+)\)$/s);
  if (!match) return;
  const inner = match[1], parts = splitOutsideParens(inner);
  if (parts.length < 2) return;
  let direction, startIdx = 0;
  const firstPart = parts[0].trim();
  (firstPart.startsWith("to ") || /^\d+(\.\d+)?(deg|rad|turn|grad)$/.test(firstPart)) && (direction = firstPart, startIdx = 1);
  const colorStops = [];
  for (let i = startIdx; i < parts.length; i++) {
    const stopParts = parts[i].trim().split(/\s+/), colorRaw = stopParts[0], color = resolveColor(colorRaw, tokenMap), positions = stopParts.slice(1), stop = { color };
    positions.length > 0 && (stop.positions = positions), colorStops.push(stop);
  }
  const gradient = {
    type: "linear-gradient",
    colorStops
  };
  return direction && (gradient.direction = direction), [gradient];
}
function parseBoxShadow(css, tokenMap) {
  const shadowStrings = splitOutsideParens(css), shadows = [];
  for (const raw of shadowStrings) {
    const s = raw.trim();
    if (!s) continue;
    const tokens = s.split(/\s+/);
    if (tokens.length < 2) return;
    let startIdx = 0, inset = !1;
    tokens[0] === "inset" && (inset = !0, startIdx = 1);
    const numericParts = [];
    let colorParts = [];
    for (let i = startIdx; i < tokens.length; i++) {
      const n = parseDimension(tokens[i]);
      if (n !== void 0)
        numericParts.push(n);
      else {
        colorParts = tokens.slice(i);
        break;
      }
    }
    if (numericParts.length < 2) return;
    const shadow = {
      offsetX: numericParts[0],
      offsetY: numericParts[1]
    };
    if (inset && (shadow.inset = !0), numericParts.length >= 3 && (shadow.blurRadius = numericParts[2]), numericParts.length >= 4 && (shadow.spreadDistance = numericParts[3]), colorParts.length > 0) {
      const colorStr = colorParts.join(" ");
      shadow.color = resolveColor(colorStr, tokenMap);
    }
    shadows.push(shadow);
  }
  return shadows.length > 0 ? shadows : void 0;
}
function parseTextShadow(css, tokenMap) {
  const tokens = css.trim().split(/\s+/);
  if (tokens.length < 3) return;
  const offsetX = parseDimension(tokens[0]), offsetY = parseDimension(tokens[1]), blur = parseDimension(tokens[2]);
  if (offsetX === void 0 || offsetY === void 0 || blur === void 0)
    return;
  const result = [
    ["textShadowOffset", { width: offsetX, height: offsetY }],
    ["textShadowRadius", blur]
  ];
  if (tokens.length >= 4) {
    const colorStr = tokens.slice(3).join(" ");
    result.push(["textShadowColor", resolveColor(colorStr, tokenMap)]);
  }
  return result;
}
function parseDimension(s) {
  const cleaned = s.replace(/px$|dp$/, ""), n = Number(cleaned);
  return Number.isFinite(n) ? n : void 0;
}
function splitOutsideParens(s) {
  const parts = [];
  let depth = 0, start = 0;
  for (let i = 0; i < s.length; i++) {
    const ch = s.charCodeAt(i);
    ch === 40 ? depth++ : ch === 41 ? depth-- : ch === 44 && depth === 0 && (parts.push(s.slice(start, i)), start = i + 1);
  }
  return parts.push(s.slice(start)), parts;
}
export {
  parseNativeStyle
};
