// @tamagui/web@1.144.4/dist/esm/helpers/parseNativeStyle.native.js
function parseNativeStyle(key, cssString, tokenMap) {
  switch (key) {
    case "backgroundImage":
      return parseBackgroundImage(cssString, tokenMap);
    case "boxShadow":
      return parseBoxShadow(cssString, tokenMap);
    case "textShadow":
      return parseTextShadow(cssString, tokenMap);
    default:
      return;
  }
}
function resolveColor(raw, tokenMap) {
  return tokenMap && tokenMap.has(raw) ? tokenMap.get(raw) : raw;
}
function parseBackgroundImage(css, tokenMap) {
  var match = css.match(RegExp("^linear-gradient\\((.+)\\)$", "s"));
  if (match) {
    var inner = match[1],
      parts = splitOutsideParens(inner);
    if (!(parts.length < 2)) {
      var direction,
        startIdx = 0,
        firstPart = parts[0].trim();
      (firstPart.startsWith("to ") || /^\d+(\.\d+)?(deg|rad|turn|grad)$/.test(firstPart)) && (direction = firstPart, startIdx = 1);
      for (var colorStops = [], i = startIdx; i < parts.length; i++) {
        var stopParts = parts[i].trim().split(/\s+/),
          colorRaw = stopParts[0],
          color = resolveColor(colorRaw, tokenMap),
          positions = stopParts.slice(1),
          stop = {
            color
          };
        positions.length > 0 && (stop.positions = positions), colorStops.push(stop);
      }
      var gradient = {
        type: "linear-gradient",
        colorStops
      };
      return direction && (gradient.direction = direction), [gradient];
    }
  }
}
function parseBoxShadow(css, tokenMap) {
  var shadowStrings = splitOutsideParens(css),
    shadows = [],
    _iteratorNormalCompletion = !0,
    _didIteratorError = !1,
    _iteratorError = void 0;
  try {
    for (var _iterator = shadowStrings[Symbol.iterator](), _step; !(_iteratorNormalCompletion = (_step = _iterator.next()).done); _iteratorNormalCompletion = !0) {
      var raw = _step.value,
        s = raw.trim();
      if (s) {
        var tokens = s.split(/\s+/);
        if (tokens.length < 2) return;
        var startIdx = 0,
          inset = !1;
        tokens[0] === "inset" && (inset = !0, startIdx = 1);
        for (var numericParts = [], colorParts = [], i = startIdx; i < tokens.length; i++) {
          var n = parseDimension(tokens[i]);
          if (n !== void 0) numericParts.push(n);else {
            colorParts = tokens.slice(i);
            break;
          }
        }
        if (numericParts.length < 2) return;
        var shadow = {
          offsetX: numericParts[0],
          offsetY: numericParts[1]
        };
        if (inset && (shadow.inset = !0), numericParts.length >= 3 && (shadow.blurRadius = numericParts[2]), numericParts.length >= 4 && (shadow.spreadDistance = numericParts[3]), colorParts.length > 0) {
          var colorStr = colorParts.join(" ");
          shadow.color = resolveColor(colorStr, tokenMap);
        }
        shadows.push(shadow);
      }
    }
  } catch (err) {
    _didIteratorError = !0, _iteratorError = err;
  } finally {
    try {
      !_iteratorNormalCompletion && _iterator.return != null && _iterator.return();
    } finally {
      if (_didIteratorError) throw _iteratorError;
    }
  }
  return shadows.length > 0 ? shadows : void 0;
}
function parseTextShadow(css, tokenMap) {
  var tokens = css.trim().split(/\s+/);
  if (!(tokens.length < 3)) {
    var offsetX = parseDimension(tokens[0]),
      offsetY = parseDimension(tokens[1]),
      blur = parseDimension(tokens[2]);
    if (!(offsetX === void 0 || offsetY === void 0 || blur === void 0)) {
      var result = [["textShadowOffset", {
        width: offsetX,
        height: offsetY
      }], ["textShadowRadius", blur]];
      if (tokens.length >= 4) {
        var colorStr = tokens.slice(3).join(" ");
        result.push(["textShadowColor", resolveColor(colorStr, tokenMap)]);
      }
      return result;
    }
  }
}
function parseDimension(s) {
  var cleaned = s.replace(/px$|dp$/, ""),
    n = Number(cleaned);
  return Number.isFinite(n) ? n : void 0;
}
function splitOutsideParens(s) {
  for (var parts = [], depth = 0, start = 0, i = 0; i < s.length; i++) {
    var ch = s.charCodeAt(i);
    ch === 40 ? depth++ : ch === 41 ? depth-- : ch === 44 && /* , */
    depth === 0 && (parts.push(s.slice(start, i)), start = i + 1);
  }
  return parts.push(s.slice(start)), parts;
}
export { parseNativeStyle };
