// @tamagui/sheet@1.144.4/dist/cjs/useGestureHandlerPan.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: !0 });
}, __copyProps = (to, from, except, desc) => {
  if (from && typeof from == "object" || typeof from == "function")
    for (let key of __getOwnPropNames(from))
      !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: !0 }), mod);
var useGestureHandlerPan_exports = {};
__export(useGestureHandlerPan_exports, {
  useGestureHandlerPan: () => useGestureHandlerPan
});
module.exports = __toCommonJS(useGestureHandlerPan_exports);
var import_react = require("react"), import_gestureState = require("./gestureState");
const AT_TOP_THRESHOLD = 5;
function useGestureHandlerPan(config) {
  const {
    positions,
    frameSize,
    setPosition,
    animateTo,
    stopSpring,
    scrollBridge,
    setIsDragging,
    getCurrentPosition,
    resisted,
    disableDrag,
    isShowingInnerSheet,
    setAnimatedPosition,
    scrollGestureRef
  } = config, gestureHandlerEnabled = (0, import_gestureState.isGestureHandlerEnabled)(), panGestureRef = (0, import_react.useRef)(null), gestureStateRef = (0, import_react.useRef)({
    startY: 0,
    // track last translation when pan was active (for position calculation after handoff)
    lastPanTranslationY: 0,
    // accumulated position offset from all pan movements
    accumulatedOffset: 0,
    // track previous translation for direction detection (like actions-sheet)
    prevTranslationY: 0,
    // track if scroll was engaged (scrollY > 0) at some point
    scrollEngaged: !1,
    // positions frozen at gesture start — keyboard may dismiss during drag (input blur),
    // causing positions to revert. Frozen positions ensure stable snap calculation.
    frozenPositions: [],
    frozenMinY: 0,
    // whether pan gesture actually started (vs just a tap in onBegin)
    panStarted: !1
  }), onStart = (0, import_react.useCallback)(() => {
    stopSpring();
  }, [stopSpring]), onEnd = (0, import_react.useCallback)(
    (closestPoint, animationOverride) => {
      setIsDragging(!1), scrollBridge.setParentDragging(!1), scrollBridge.setScrollEnabled?.(!0), setPosition(closestPoint), animateTo(closestPoint, animationOverride);
    },
    [setIsDragging, scrollBridge, setPosition, animateTo]
  );
  return {
    panGesture: (0, import_react.useMemo)(() => {
      if (!gestureHandlerEnabled || disableDrag || isShowingInnerSheet || !frameSize)
        return null;
      const { Gesture } = (0, import_gestureState.getGestureHandlerState)();
      if (!Gesture)
        return null;
      const minY = positions[0], gs = gestureStateRef.current, gesture = Gesture.Pan().withRef(panGestureRef).activeOffsetY([-10, 10]).failOffsetX([-20, 20]).shouldCancelWhenOutside(!1).onBegin(() => {
        gs.panStarted = !1, config.pauseKeyboardHandler && (config.pauseKeyboardHandler.current = !0);
        const pos = getCurrentPosition(), atTop = pos <= minY + AT_TOP_THRESHOLD, currentScrollY = scrollBridge.y;
        gs.startY = pos, gs.lastPanTranslationY = 0, gs.accumulatedOffset = 0, gs.prevTranslationY = 0, gs.scrollEngaged = currentScrollY > 0, gs.frozenPositions = [...positions], gs.frozenMinY = minY, atTop || scrollBridge.setScrollEnabled?.(!1, 0);
      }).onStart(() => {
        gs.panStarted = !0, setIsDragging(!0), scrollBridge.initialPosition = gs.startY, onStart();
      }).onChange((event) => {
        const { translationY } = event, isSwipingDown = gs.prevTranslationY < translationY, deltaY = translationY - gs.prevTranslationY;
        gs.prevTranslationY = translationY;
        const scrollY = scrollBridge.y;
        scrollY > 0 && (gs.scrollEngaged = !0);
        const isCurrentlyAtTop = gs.startY + gs.accumulatedOffset <= minY + AT_TOP_THRESHOLD, nodeIsScrolling = scrollY > 0;
        let panHandles = !1;
        const hasScrollableContent = scrollBridge.hasScrollableContent !== !1;
        if (isCurrentlyAtTop ? isSwipingDown ? nodeIsScrolling && hasScrollableContent ? panHandles = !1 : (gs.scrollEngaged, panHandles = !0) : hasScrollableContent ? panHandles = !1 : panHandles = !0 : isSwipingDown ? panHandles = !nodeIsScrolling || !hasScrollableContent : panHandles = !0, panHandles) {
          const lockTo = isCurrentlyAtTop ? void 0 : 0;
          scrollBridge.setScrollEnabled?.(!1, lockTo), gs.accumulatedOffset += deltaY;
          const newPosition = resisted(gs.startY + gs.accumulatedOffset, minY);
          scrollBridge.paneY = newPosition, setAnimatedPosition(newPosition), scrollBridge.setParentDragging(newPosition > minY);
        } else
          scrollBridge.setScrollEnabled?.(!0);
      }).onEnd((event) => {
        const { velocityY } = event, currentPos = gs.startY + gs.accumulatedOffset;
        scrollBridge.scrollLockY = void 0;
        const snapPositions = gs.frozenPositions.length > 0 ? gs.frozenPositions : positions, snapMinY = gs.frozenPositions.length > 0 ? gs.frozenMinY : minY;
        if (currentPos <= snapMinY + AT_TOP_THRESHOLD && scrollBridge.y > 0) {
          onEnd(0);
          return;
        }
        const velocity = velocityY / 1e3, projectedEnd = currentPos + frameSize * velocity * 0.2;
        let closestPoint = 0, minDist = Number.POSITIVE_INFINITY;
        for (let i = 0; i < snapPositions.length; i++) {
          const pos = snapPositions[i], dist = Math.abs(projectedEnd - pos);
          dist < minDist && (minDist = dist, closestPoint = i);
        }
        onEnd(closestPoint);
      }).onFinalize(() => {
        scrollBridge.scrollLockY = void 0, gs.panStarted ? setIsDragging(!1) : config.pauseKeyboardHandler && (config.pauseKeyboardHandler.current = !1);
      }).runOnJS(!0);
      return scrollGestureRef?.current ? gesture.simultaneousWithExternalGesture(scrollGestureRef.current) : gesture;
    }, [
      gestureHandlerEnabled,
      disableDrag,
      isShowingInnerSheet,
      frameSize,
      positions,
      scrollBridge,
      getCurrentPosition,
      resisted,
      onStart,
      onEnd,
      setIsDragging,
      setAnimatedPosition
    ]),
    panGestureRef,
    gestureHandlerEnabled
  };
}
