// @tamagui/sheet@1.144.4/dist/cjs/useKeyboardControllerSheet.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: !0 });
}, __copyProps = (to, from, except, desc) => {
  if (from && typeof from == "object" || typeof from == "function")
    for (let key of __getOwnPropNames(from))
      !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: !0 }), mod);
var useKeyboardControllerSheet_exports = {};
__export(useKeyboardControllerSheet_exports, {
  useKeyboardControllerSheet: () => useKeyboardControllerSheet
});
module.exports = __toCommonJS(useKeyboardControllerSheet_exports);
var import_react = require("react");
const noop = () => {
};
function useKeyboardControllerSheet(_options) {
  const pauseKeyboardHandler = (0, import_react.useRef)(!1);
  return {
    keyboardControllerEnabled: !1,
    keyboardHeight: 0,
    isKeyboardVisible: !1,
    dismissKeyboard: noop,
    pauseKeyboardHandler,
    flushPendingHide: noop
  };
}
