// @tamagui/sheet@1.144.4/dist/cjs/useSheetScrollViewGestures.js
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: !0 });
}, __copyProps = (to, from, except, desc) => {
  if (from && typeof from == "object" || typeof from == "function")
    for (let key of __getOwnPropNames(from))
      !__hasOwnProp.call(to, key) && key !== except && __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: !0 }), mod);
var useSheetScrollViewGestures_exports = {};
__export(useSheetScrollViewGestures_exports, {
  useSheetScrollViewGestures: () => useSheetScrollViewGestures
});
module.exports = __toCommonJS(useSheetScrollViewGestures_exports);
var import_react = require("react");
function useSheetScrollViewGestures({
  scrollRef,
  scrollBridge,
  hasScrollableContent
}) {
  const state = (0, import_react.useRef)({
    startY: 0,
    lastY: 0,
    owner: "none",
    panDragOffset: 0,
    dys: [],
    scrollYAtGestureStart: 0
  });
  return (0, import_react.useEffect)(() => {
    if (!scrollRef.current) return;
    const controller = new AbortController(), node = scrollRef.current?.getScrollableNode();
    if (!node) return;
    node.style.overscrollBehavior = "contain";
    let originalOverflow = "";
    const disableScroll = () => {
      node.style.overflowY !== "hidden" && (originalOverflow = node.style.overflowY, node.style.overflowY = "hidden");
    }, enableScroll = () => {
      node.style.overflowY = originalOverflow;
    }, handleTouchStart = (e) => {
      const touch = e.touches[0];
      if (!touch) return;
      const currentScrollY = node.scrollTop;
      state.current = {
        startY: touch.pageY,
        lastY: touch.pageY,
        owner: "none",
        panDragOffset: 0,
        dys: [],
        scrollYAtGestureStart: currentScrollY
      }, scrollBridge.scrollStartY = touch.pageY;
    }, handleTouchMove = (e) => {
      const touch = e.touches[0];
      if (!touch) return;
      const { current: s } = state, pageY = touch.pageY, dy = pageY - s.lastY;
      if (s.lastY = pageY, dy === 0) return;
      const isDraggingDown = dy > 0, currentScrollY = node.scrollTop, isPaneAtTop = scrollBridge.paneY <= scrollBridge.paneMinY + 5;
      scrollBridge.y = currentScrollY;
      let shouldPanOwn = !1;
      isPaneAtTop ? currentScrollY <= 0 && isDraggingDown && (shouldPanOwn = !0) : shouldPanOwn = !0;
      const newOwner = shouldPanOwn ? "pan" : "scroll";
      if (newOwner !== s.owner && (newOwner === "pan" ? (s.panDragOffset = 0, s.dys = [], scrollBridge.setParentDragging(!0), disableScroll()) : (scrollBridge.setParentDragging(!1), scrollBridge.scrollLock = !1, enableScroll()), s.owner = newOwner), s.owner === "pan")
        e.cancelable && e.preventDefault(), s.panDragOffset += dy, scrollBridge.drag(s.panDragOffset), s.dys.push(dy), s.dys.length > 100 && (s.dys = s.dys.slice(-10));
      else if (s.owner === "scroll") {
        const scrollDelta = -dy, maxScrollY = node.scrollHeight - node.clientHeight, newScrollY = Math.max(0, Math.min(maxScrollY, currentScrollY + scrollDelta));
        newScrollY !== currentScrollY && (node.scrollTop = newScrollY, scrollBridge.y = newScrollY);
      }
    }, handleTouchEnd = () => {
      const { current: s } = state;
      if (s.owner === "pan") {
        scrollBridge.setParentDragging(!1);
        let vy = 0;
        if (s.dys.length) {
          const recentDys = s.dys.slice(-10);
          vy = recentDys.reduce((a, b) => a + b, 0) / recentDys.length * 0.04;
        }
        scrollBridge.release({ dragAt: s.panDragOffset, vy });
      }
      enableScroll(), s.owner = "none", s.panDragOffset = 0, s.dys = [];
    };
    return node.addEventListener("touchstart", handleTouchStart, {
      signal: controller.signal,
      passive: !0
    }), node.addEventListener("touchmove", handleTouchMove, {
      signal: controller.signal,
      passive: !1
    }), node.addEventListener("touchend", handleTouchEnd, {
      signal: controller.signal,
      passive: !0
    }), node.addEventListener("touchcancel", handleTouchEnd, {
      signal: controller.signal,
      passive: !0
    }), () => {
      enableScroll(), controller.abort();
    };
  }, [scrollRef, scrollBridge, hasScrollableContent]), {};
}
