// @tamagui/image@1.144.4/dist/esm/createImage.native.js
import { jsx as _jsx } from "react/jsx-runtime";
import { getTokenValue, styled } from "@tamagui/web";
var defaultObjectFitMap = function (objectFit) {
    switch (objectFit) {
      case "fill":
        return "stretch";
      case "none":
        return "center";
      case "scale-down":
        return "contain";
      case "contain":
        return "contain";
      default:
        return "cover";
    }
  },
  defaultTransformSource = function (props) {
    var {
      src,
      source,
      width,
      height
    } = props;
    return source || (src && typeof src != "string" ? src : {
      uri: src,
      width,
      height
    });
  };
function createImage(options) {
  var {
      Component,
      mapObjectFitToResizeMode = defaultObjectFitMap,
      resizeModePropName = "resizeMode",
      objectPositionPropName,
      transformSource = defaultTransformSource
    } = options,
    inlinePropsSet = /* @__PURE__ */new Set(["source", "placeholder", "transition", "contentFit", "contentPosition", "cachePolicy", "recyclingKey", "allowDownscaling", "autoplay", "blurRadius", "priority", "placeholderContentFit", "responsivePolicy", "onLoadStart", "onProgress", "onLoadEnd"]),
    StyledImage = styled(Component, {
      name: "Image"
    }, {
      inlineProps: inlinePropsSet
    }),
    ImageComponent = StyledImage.styleable(function (incomingProps, ref) {
      var props = incomingProps,
        {
          src,
          width,
          height,
          objectFit,
          objectPosition,
          // web only props - filter out on native
          decoding,
          elementTiming,
          fetchpriority,
          isMap,
          loading,
          sizes,
          useMap,
          onLoad,
          onError,
          ...rest
        } = props,
        resolvedWidth = typeof width == "string" && width[0] === "$" ? getTokenValue(width) : width,
        resolvedHeight = typeof height == "string" && height[0] === "$" ? getTokenValue(height) : height,
        finalSource = transformSource({
          src,
          width: resolvedWidth,
          height: resolvedHeight
        }),
        finalProps = {
          ...rest,
          source: finalSource,
          style: {
            width: resolvedWidth,
            height: resolvedHeight
          }
        };
      return objectFit && (finalProps[resizeModePropName] = mapObjectFitToResizeMode(objectFit)), objectPositionPropName && objectPosition && (finalProps[objectPositionPropName] = objectPosition), onLoad && (finalProps.onLoad = function (e) {
        var _e_nativeEvent,
          source = (e == null || (_e_nativeEvent = e.nativeEvent) === null || _e_nativeEvent === void 0 ? void 0 : _e_nativeEvent.source) || e?.source || {};
        onLoad({
          target: {
            naturalHeight: source?.height,
            naturalWidth: source?.width
          },
          type: "load"
        });
      }), onError && (finalProps.onError = function () {
        onError({
          type: "error"
        });
      }), /* @__PURE__ */_jsx(Component, {
        ref,
        ...finalProps
      });
    }),
    comp = Component;
  return ImageComponent.getSize = comp.getSize || function () {}, ImageComponent.getSizeWithHeaders = comp.getSizeWithHeaders || function () {}, ImageComponent.prefetch = comp.prefetch || function () {}, ImageComponent.prefetchWithMetadata = comp.prefetchWithMetadata || function () {}, ImageComponent.abortPrefetch = comp.abortPrefetch || function () {}, ImageComponent.queryCache = comp.queryCache || function () {}, ImageComponent;
}
export { createImage };
